www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,409 @@
+    close: () => setVisible(false),
+  }));
+  const commodityOptions = useMemo(() => {
+    return commodityStore.list.map((comm) => ({
+      value: comm.uri,
+      label: comm.name,
+    }));
+  }, [commodityStore.records.size]);
+  const countryOptions = useMemo(() => {
+    return countryStore.list.map((country) => ({
+      value: country.uri,
+      label: country.name,
+    }));
+  }, [countryStore.records.size]);
+
+  const stateOptions = useMemo(() => {
+    return stateOrProvinceStore.list.map((state) => ({
+      value: state.uri,
+      label: state.name,
+    }));
+  }, [stateOrProvinceStore.records.size]);
+
+  const depositTypeOptions = useMemo(() => {
+    return depositTypeStore.list.map((type) => ({
+      value: type.uri,
+      label: type.name,
+    }));
+  }, [depositTypeStore.records.size]);
+
+  const unitOptions = useMemo(() => {
+    return unitStore.list.map((unit) => ({
+      value: unit.uri,
+      label: unit.name,
+    }));
+  }, [unitStore.records.size]);
+
+  // we need source type and reference to update the source id
+  const handleSourceTypeChange = (e: RadioChangeEvent) => {
+    const value = e.target.value;
+    setSelectedSourceType(value);
+  };
+  const handleSave = async (values: FormValues) => {
+    const currentUserUrl = userStore.getCurrentUser()!.url;
+    let location = undefined;
+    if (values.latitude !== undefined && values.longitude !== undefined) {
+      location = `POINT (${values.longitude} ${values.latitude})`;
+    }
+
+    const countries = values.country
+      ? [
+        new CandidateEntity({
+          observedName: countryStore.getByURI(values.country)!.name,
+          source: currentUserUrl,
+          normalizedURI: values.country,
+          confidence: 1.0,
+        }),
+      ]
+      : [];
+
+    const statesOrProvinces = values.stateOrProvince
+      ? [
+        new CandidateEntity({
+          observedName: stateOrProvinceStore.getByURI(values.stateOrProvince)!.name,
+          source: currentUserUrl,
+          normalizedURI: values.stateOrProvince,
+          confidence: 1.0,
+        }),
+      ]
+      : [];
+
+    const commodity1 = commodity?.id;
+    const referenceDocument = new Document({
+      uri: values.refDoc,
+    });
+
+    const reference = new Reference({
+      document: referenceDocument,
+      comment: values.refComment,
+    });
+
+    const mineralInventory = new MineralInventory({
+      category: ["Inferred", "Indicated", "Measured"].map(
+        (name) =>
+          new CandidateEntity({
+            source: currentUserUrl,
+            confidence: 1.0,
+            observedName: name,
+            normalizedURI: `https://minmod.isi.edu/resource/${name}`,
+          })
+      ),
+      commodity: new CandidateEntity({
+        source: currentUserUrl,
+        confidence: 1.0,
+        observedName: commodity?.name,
+        normalizedURI: commodity?.uri,
+      }),
+      grade: values.grade !== undefined && values.grade !== null
```

Comment:
values.grade can only be undefined or a number based on typing, no need to compare with null.